packages/bee-q/src/components/tab-group/bq-tab-group.tsx

```diff
@@ -0,0 +1,198 @@
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** A string representing the id of the selected tab. */
+  @Prop({ reflect: true, mutable: true }) value: string;
+
+  /** The size of the tab */
+  @Prop({ reflect: true }) size: TTabSize = 'small';
+
+  /** A number representing the delay value applied to bqChange event handler */
+  @Prop({ reflect: true, mutable: true }) debounceTime = 0;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('debounceTime')
+  checkDebounceChange() {
+    if (this.debounceTime < 0) {
+      this.debounceTime = Math.max(0, this.debounceTime);
+    }
+
+    if (this.debouncedBqChange) {
+      this.debouncedBqChange.cancel();
+    }
+
+    this.debouncedBqChange = debounce((event: Parameters<typeof this.debouncedBqChange>[0]) => {
+      this.bqChange.emit(event);
+    }, this.debounceTime);
+  }
+
+  @Watch('size')
+  checkPropValues() {
+    validatePropValue(TAB_SIZE, 'small', this.el, 'size');
+
+    this.bqTabElements.forEach((bqTabElement) => {
+      bqTabElement.size = this.size;
+      bqTabElement.divider = true;
+    });
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when the tab value changes */
+  @Event() bqChange: EventEmitter<{ target: HTMLBqTabElement; value: string }>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.checkDebounceChange();
+    this.checkPropValues();
+  }
+
+  componentDidLoad() {
+    this.bqTabElements.forEach((bqTabElement) => {
+      bqTabElement.active = !isNil(this.value) ? bqTabElement.tabId === this.value : false;
+      bqTabElement.size = this.size;
+    });
+  }
+
+  // Listeners
+  // ==============
+
+  @Listen('mousedown', { target: 'body', passive: true })
+  onMouseDown(event: MouseEvent) {
+    if (!isNil(this.focusedBqTab) && isHTMLElement(event.target, 'bq-tab') && this.el.contains(event.target)) {
+      this.focusedBqTab = event.target;
+    }
+  }
+
+  @Listen('bqClick')
+  onBqClick(event: CustomEvent<HTMLBqTabElement>) {
+    const { detail: target } = event;
+    this.bqTabElements.forEach((bqTabElement) => (bqTabElement.active = bqTabElement === target));
+    this.debouncedBqChange({ value: target.tabId, target });
+    this.selectTab(target);
+  }
+
+  @Listen('bqKeyDown')
+  onBqKeyDown(event: CustomEvent<KeyboardEvent>) {
+    const { target } = event;
+
+    if (!isHTMLElement(target, 'bq-tab')) return;
+
+    switch (event.detail.key) {
+      case 'ArrowRight': {
+        this.focusTabSibbling(target, 'forward');
+        break;
+      }
+      case 'ArrowLeft': {
+        this.focusTabSibbling(target, 'backward');
+        break;
+      }
+      default:
+    }
+  }
```

Comment:
We should allow regular focus as well. I mean is good to change the selected Tab while pressing the Left/Right arrow key, but the user should be able to focus using the Tab key as well, without activating the focused Tab... Eg:

https://user-images.githubusercontent.com/328492/226837751-eca44e0f-0c0c-4b94-b3b2-38ad1df59b8c.mp4

Reply:
That makes sense, updated.